### PR TITLE
Allow to bypass the oauth loading page

### LINKED ISSUE #2109: Skip the user login page when OIDC is used
**Question/Feature request**
Is there any way to skip the user login page when OIDC login is enabled? We're integrating kubeapp start link in another dashboard and the user is already logged via OIDC in the parent dashboard. Something similar to Harbor https://github.com/goharbor/harbor/pull/9311

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

As per #2109, allow users to configure Kubeapps so it skips the login/welcome page and redirect directly to the OAuth login URI.

To use it, a value is exposed in the chart: `--set authProxy.skipKubeappsLoginPage=true`.

### Applicable issues

  - fixes #2109

### CODE REVIEW

File: chart/kubeapps/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: kubeapps
-version: 4.0.4
+version: 4.0.5
```
Comment (absoludity) [inline chart/kubeapps/Chart.yaml:3]: Maybe land this after @antgamdia 's PR with the ClusterRole, so they're both included in the chart update (once you've reviewed his).

File: dashboard/src/actions/auth.ts [hunk 2]
```diff
@@ -99,8 +99,10 @@ export function checkCookieAuthentication(
     const isAuthed = await Auth.isAuthenticatedWithCookie(cluster);
     if (isAuthed) {
       await dispatch(authenticate(cluster, "", true));
+      return true;
     } else {
       dispatch(setAuthenticated(false, false, ""));
+      return false;
     }
```
Comment (absoludity) [inline dashboard/src/actions/auth.ts:106]: Perhaps just 
```
return isAuthed;
```
instead?

File: dashboard/src/components/LoginForm/LoginForm.test.tsx [hunk 3]
```diff
@@ -25,31 +28,16 @@ const defaultProps = {
...
-    mount(<LoginForm {...props} checkCookieAuthentication={checkCookieAuthentication} />);
-    expect(checkCookieAuthentication).toHaveBeenCalled();
-  });
-
-  it("does not call checkCookieAuthentication when oauthLoginURI not provided", () => {
-    const checkCookieAuthentication = jest.fn();
-    mount(<LoginForm {...defaultProps} checkCookieAuthentication={checkCookieAuthentication} />);
-    expect(checkCookieAuthentication).not.toHaveBeenCalled();
-  });
-});
```
Comment (absoludity) [inline dashboard/src/components/LoginForm/LoginForm.test.tsx:51]: Why are these removed?

**Review by absoludity — APPROVED** [review]
Nice, thanks.

File: chart/kubeapps/Chart.yaml (hunk 1 above)
Comment (andresmgot) [reply chart/kubeapps/Chart.yaml:3]: well, @antgamdia changes cannot be used until the new endpoint (`/refresh`) is available so the change in the chart is not going to be usable yet.

File: dashboard/src/components/LoginForm/LoginForm.test.tsx (hunk 3 above)
Comment (andresmgot) [reply dashboard/src/components/LoginForm/LoginForm.test.tsx:51]: Because this was already been covered in the tests under "oauth login form" (sorry I forgot to mention that).
